numpy/core/src/multiarray/alloc.c

```diff
@@ -113,8 +113,11 @@ npy_free_cache(void * p, npy_uintp sz)
 NPY_NO_EXPORT void *
 npy_alloc_cache_dim(npy_uintp sz)
 {
-    /* dims + strides */
-    if (NPY_UNLIKELY(sz < 2)) {
+    /*
+     * make sure any temporary allocation can be used for array metadata which
+     * uses one memory block for both dimensions and strides
+     */
```

Comment:
Still don't follow here - how do we end up on this code path? To me, it seems like we always passed an even number in the past.

So the only situation is when `sz == 0`. Why do we allocate any memory at all? Why would you ask for 0 bytes then proceed to write two of them?